libbeat/cmd/instance/beat.go

```diff
@@ -750,28 +732,26 @@ func (b *Beat) loadDashboards(ctx context.Context, force bool) error {
 	return nil
 }
 
-// registerTemplateLoading registers the loading of the template as a callback with
-// the elasticsearch output. It is important the the registration happens before
-// the publisher is created.
-func (b *Beat) registerTemplateLoading() error {
-	var templateCfg template.TemplateConfig
+// registerESIndexManagement registers the loading of the template and ILM
+// policy as a callback with the elasticsearch output. It is important the
+// registration happens before the publisher is created.
+func (b *Beat) registerESIndexManagement() error {
+	if b.Config.Output.Name() != "elasticsearch" {
+		return nil
+	}
 
-	// Check if outputting to file is enabled, and output to file if it is
+	// 1. validate template setup settings
 	if b.Config.Template.Enabled() {
+		var templateCfg template.TemplateConfig
```

Comment:
With passing the `rawConfig` object to the `idxmgmt.Supporter` you allow that everything index, template and ilm related is under full control of the supporter. By checking for the `b.Config.Template.Enabled()` directly here, the logic could diverge. Shouldn't this be done inside the supporter, e.g. by adding another method like `Validate` to the handler?

Reply:
Oh, nice one. Indeed, this is some leftover that should have been removed by now. The Supporter should be responsible only. The second block calling `b.index.Enabled()` is enough to decide whether to register the callback or not. The config validation must be handled by the Supporter on init. Will remove it and the actual Config.Template.

In the future I want to remove the global callback, and have the output decide if it can make use of index support. The output will need to implement an APIHanlder only. This might allow us to integrate index setup via Beats->Logstash->Elasticsearch.
But I refrained from doing so, so to not interfere other potential users of the global callback.